rust/fastsim-core/src/traits.rs

```diff
@@ -35,62 +35,66 @@ pub trait SerdeAPI: Serialize + for<'a> Deserialize<'a> {
     ///
     /// # Argument:
     ///
-    /// * `filename`: a `str` storing the targeted file name. Currently `.json` and `.yaml` suffixes are
+    /// * `filepath`: a `str` storing the targeted file name. Currently `.json` and `.yaml` suffixes are
     /// supported
     ///
     /// # Returns:
     ///
     /// A Rust Result wrapping data structure if method is called successfully; otherwise a dynamic
     /// Error.
-    fn from_file(filename: &str) -> Result<Self, anyhow::Error>
+    fn from_file(filepath: &str) -> anyhow::Result<Self>
     where
         Self: std::marker::Sized,
         for<'de> Self: Deserialize<'de>,
     {
-        let extension = Path::new(filename)
+        let extension = Path::new(filepath)
             .extension()
             .and_then(OsStr::to_str)
             .unwrap_or("");
 
-        let file = File::open(filename)?;
+        let file = File::open(filepath)?;
         // deserialized file
         let mut file_de: Self = match extension {
             "yaml" => serde_yaml::from_reader(file)?,
             "json" => serde_json::from_reader(file)?,
-            _ => bail!("Unsupported file extension {}", extension),
+            _ => anyhow::bail!("Unsupported file extension {}", extension),
         };
         file_de.init()?;
         Ok(file_de)
     }
 
-    /// json serialization method.
-    fn to_json(&self) -> String {
-        serde_json::to_string(&self).unwrap()
+    // /// JSON serialization method
```

Comment:
@kylecarow , what's going on with this comment block?

Reply:
@calbaker oops, that's leftover from my debugging, I'll remove it